python/lsst/pipe/base/pipeline.py

```diff
@@ -189,9 +191,43 @@ def fromFile(cls, filename: str) -> Pipeline:
         string based matching due to the nature of contracts and may prune more
         than it should.
         """
-        # Split up the filename and any labels that were supplied
-        filename, labelSpecifier = cls._parseFileSpecifier(filename)
-        pipeline: Pipeline = cls.fromIR(pipelineIR.PipelineIR.from_file(filename))
+        return cls.fromUri(filename)
+
+    @classmethod
+    def fromUri(cls, uri):
+        """Load a pipeline defined in a pipeline yaml file at a location
+        specified by a URI.
+
+        Parameters
+        ----------
+        uri: `str`
```

Comment:
I think most other middleware methods that take a URI actually take a `ButlerURI` instance instead of or in addition to a `str`.  And I _bet_ `ButlerURI`'s constructor will happily accept a `ButlerURI` instance, so this may just be a matter of updating the annotations and docs (but please don't take my word for it).